### PR TITLE
K8SPS-210: Add unit tests job to GA.

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPS-210](https://jira.percona.com/browse/K8SPS-210) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPS-210

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/sidecars.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/limits.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/one-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/users.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/a83c3b6/tls-cert-manager.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/233/commits/a83c3b68cb0386049f8fbae06f69a229cd00cbd1
image: `perconalab/percona-server-mysql-operator:PR-233-a83c3b6`

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Run tests
```
Comment (hors) [inline .github/workflows/tests.yml:1]:
```suggestion
name: Run Unit tests
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19'
+      - uses: actions/checkout@v3
+      - name: test
+        run: go test -v ./...
```
Comment (hors) [inline .github/workflows/tests.yml:13]: @inelpandzic please check results

**Review by hors — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (inelpandzic) [reply .github/workflows/tests.yml:13]: These unit tests are currently failing, need to fix them.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (inelpandzic) [reply .github/workflows/tests.yml:1]: We can (and will) run integration tests here as well, that's why I didn't want to put `unit` in there.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (inelpandzic) [reply .github/workflows/tests.yml:13]: [e3e1bf9](https://github.com/percona/percona-server-mysql-operator/pull/233/commits/e3e1bf94d9acdd0dcc27e13cdb49046b41053f4d)

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/sidecars.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/one-pod.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/haproxy.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/e3e1bf9/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/233/commits/e3e1bf94d9acdd0dcc27e13cdb49046b41053f4d
image: `perconalab/percona-server-mysql-operator:PR-233-e3e1bf9`

File: controllers/suite_test.go [hunk 3]
```diff
@@ -20,13 +20,13 @@ import (
...
 
-	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
-	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	//"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
```
Comment (egegunes) [inline controllers/suite_test.go:29]: let's remove this if it causes any problems, don't commit commented code

File: go.mod [hunk 4]
```diff
@@ -14,7 +14,8 @@ require (
 	github.com/go-openapi/validate v0.22.0
 	github.com/go-sql-driver/mysql v1.7.0
 	github.com/minio/minio-go/v7 v7.0.45
-	github.com/onsi/ginkgo v1.16.5
+	// github.com/onsi/ginkgo v1.16.5
```
Comment (egegunes) [inline go.mod:17]: why we're commenting this?

**Review by egegunes — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19'
+      - uses: actions/checkout@v3
+      - name: test
```
Comment (egegunes) [inline .github/workflows/tests.yml:12]:
```suggestion
      - name: go test
```

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/sidecars.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/one-pod.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/haproxy.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-233/68eecfd/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/233/commits/68eecfda21f6cf363dbae75b7284dee3b433b3d2
image: `perconalab/percona-server-mysql-operator:PR-233-68eecfd`

**Review by hors — APPROVED** [review]

**Pull request merged**